ompi/communicator/communicator.h

```diff
@@ -261,20 +262,30 @@ struct ompi_communicator_t {
     opal_infosubscriber_t      super;
     opal_mutex_t               c_lock; /* mutex for name and potentially
                                           attributes */
-    char  c_name[MPI_MAX_OBJECT_NAME];
+    char                      *c_name;
     ompi_comm_extended_cid_t      c_contextid;
     ompi_comm_extended_cid_block_t c_contextidb;
     uint32_t                      c_index;
     int                           c_my_rank;
     uint32_t                      c_flags; /* flags, e.g. intercomm,
                                               topology, etc. */
     uint32_t                      c_assertions; /* info assertions */
-    int c_id_available; /* the currently available Cid for allocation
-               to a child*/
-    int c_id_start_index; /* the starting index of the block of cids
-                 allocated to this communicator*/
+#if OPAL_ENABLE_FT_MPI
```

Comment:
Sorry I am catching this up after the PR has been merged, but this change makes the version compiled w/o FT non-binary compatible with the with-ft version, that's why it was non-conditional before. Let's discuss here, I can make a PR to fix it if you want.

Reply:
How does it make it non-binary compatible ? All uses of the ompi_communicator_t is inside the MPI library, and for them all ompi_communicator_t are allocated using the correct size.

Reply:
I double checked and this change is fine, because the reordering problem applies only to MCA component modules interfaces. There is no followup needed.